ktor-server/ktor-server-core/common/src/io/ktor/server/application/Application.kt

```diff
@@ -48,33 +48,34 @@ public class ApplicationPropertiesBuilder(
         modules.add(body)
     }
 
-    internal fun build(): ApplicationProperties =
-        ApplicationProperties(environment, modules, watchPaths, rootPath, developmentMode, parentCoroutineContext)
+    internal fun build(): ApplicationRootConfig =
+        ApplicationRootConfig(environment, modules, watchPaths, rootPath, developmentMode, parentCoroutineContext)
 }
 
 /**
- * An application config with which the application is running.
+ * Core configuration for a running server.
```

Comment:
as per this line, it looks like `ServerConfig`/`HttpServerConfig` could be a good name here, WDYT?
Though, it's only if other entities are renamed...

Reply:
If we do the Application -> Server rename it'll have to be in 4.0, in which case I'd rather remove this object because it doesn't really fit with the abstraction.  There's also the problem with there already being an `ApplicationConfig` which refers to the loaded properties.